File: backend/src/main/java/com/companymanagement/backend/service/PortfolioService.java
package com.companymanagement.backend.service;

import java.util.List;

import com.companymanagement.backend.model.Portfolio;

public interface PortfolioService {
    String createPortfolio(Portfolio portfolio);
    String updatePortfolio(Portfolio portfolio);
    String deletePortfolio(Long portfolioId);
    Portfolio getPortfolio(Long portfolioId);
    List<Portfolio> getAllPortfolios();
}

File: frontend/src/app/core/services/portfolio.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Portfolio } from '../interfaces/portfolio';

@Injectable({
  providedIn: 'root'
})
export class PortfolioService {
  
  private baseUrl = 'http://localhost:8080/investment_management_system/Portfolio';

  constructor(private http: HttpClient) { }

  getPortfolio(PortfolioID: number): Observable<Portfolio> {
    return this.http.get<Portfolio>(`${this.baseUrl}/${PortfolioID}`);
  }

  getAllPortfolio(): Observable<Portfolio[]> {
    return this.http.get<Portfolio[]>(this.baseUrl);
  }

  createPortfolio(Portfolio: Portfolio): Observable<string> {
    return this.http.post<string>(this.baseUrl, Portfolio);
  }

  updatePortfolio(Portfolio: Portfolio): Observable<string> {
    return this.http.put<string>(`${this.baseUrl}/update`, Portfolio);
  }

  deletePortfolio(PortfolioID: number): Observable<string> {
    return this.http.delete<string>(`${this.baseUrl}/${PortfolioID}`);
  }

}
